***Import and load structure***

In [1]:
from Bio.PDB import PDBParser
from pathlib import Path

pdb_path = Path("../data/raw/1LYZ.pdb")

parser = PDBParser(QUIET=True)
structure = parser.get_structure("1LYZ", pdb_path)

print("Structure loaded:", structure)

Structure loaded: <Structure id=1LYZ>


***count models***

In [2]:
models = list(structure.get_models())
print("Number of models:", len(models))

Number of models: 1


***count chains***

In [3]:
model = models[0]
chains = list(model.get_chains())

print("Chains:")
for chain in chains:
    print(chain.id)

Chains:
A


***count residues***

In [4]:
from Bio.PDB.Polypeptide import is_aa

residue_count = 0

for residue in model.get_residues():
    if is_aa(residue, standard=True):
        residue_count += 1

print("Number of amino acid residues:", residue_count)

Number of amino acid residues: 129


***count atoms***

In [5]:
atom_count = sum(1 for _ in model.get_atoms())
print("Number of atoms:", atom_count)

Number of atoms: 1102


## Structure Summary (1LYZ)

- Models: 1
- Chains: 1 (Chain A)
- Residues: 129 amino acids
- Atoms: 1102

### Interpretation
This confirms that lysozyme is a small, single-chain protein, making it suitable for structure-based protein engineering and computational design workflows.